include/aws/s3/private/s3_request.h

```diff
@@ -23,6 +24,69 @@ enum aws_s3_request_flags {
     AWS_S3_REQUEST_FLAG_ALWAYS_SEND = 0x00000004,
 };
 
+/**
+ * Information sent in the meta_request telemetry callback.
+ */
+struct aws_s3_request_metrics {
```

Comment:
should we expose extra things like num_parts, bytes_transferred?

Reply:
maybe i dont understand what `request_path_query` is, but how would it include num parts?

Reply:
For put parts, it will have the part number in it, but, for multipart get, it is not.

I'll add the part number to be easier for the user

Reply:
But, for multipart get, we were actually doing ranged-get, which is different from get parts. I guess it can be confused to the user as well.